### PR TITLE
[SPARK-41093][BUILD] Remove netty-tcnative-classes from Spark dependencyList

### PR DESCRIPTION
**Pull request by pan3793**
### What changes were proposed in this pull request?

Remove `netty-tcnative-classes` from Spark dependencyList.

### Why are the changes needed?

https://github.com/netty/netty/pull/12146 explained that `netty-tcnative-classes` should be optional, otherwise will cause https://github.com/netty/netty/issues/12101

It also not part of previous `netty-all`(fat jar)

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

CI pass.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -928,16 +924,6 @@
         <version>${netty.version}</version>
         <classifier>osx-x86_64</classifier>
       </dependency>
-      <!-- SPARK-38541: Add the explicit dependency on `netty-tcnative-classes` because the dependency on
-           `netty-tcnative-classes` is changed to `Optional` in [netty](https://github.com/netty/netty/pull/12146).
-           `netty-tcnative-classes` module is mainly used by SSL package in `netty-handler`.
```
Comment (pan3793) [inline pom.xml:933]: Netty support openSSL and JDK SSL, to enable openSSL support `netty-tcnative-classes` and platform native libs should be existed in classpath.
Only `netty-tcnative-classes` exist will cause https://github.com/netty/netty/issues/12101

**Review by LuciferYang — APPROVED** [review]
+1, happy to remove an unnecessary dependency if really unused

**Comment by pan3793** [comment]
cc @srowen

**Comment by srowen** [comment]
Sounds ok. How far back should this backport?

**Comment by pan3793** [comment]
This patch only suitable to master.

- branch-3.2 and earlier use the fat netty-all, no issue;
- branch-3.3 depends on netty 4.1.74, which claims `netty-tcnative-classes` as compile scope, we need to 1) explicitly exclude `netty-tcnative-classes` from netty, 2) or upgrade netty and apply this patch, 3) or leave it, since it's not a big deal;
- master depends on netty 4.1.75+, which has the issue described by this PR;

**Comment by srowen** [comment]
Merged to master

**Comment by attilapiros** [comment]
@pan3793  On java8 is not this change has some performance impact? Based on this comment:
https://github.com/grpc/grpc-java/issues/6997#issue-610896065

> Since we do some of our own detection, we should make sure it works with these more recent Java 8 versions. **Even though Java 8 has very poor AES GCM performance**, if it is an additional option it can help users on nicher platforms why may not need high throughput.

**Comment by pan3793** [comment]
@attilapiros Sorry, I don't have experience with SSL-enabled Spark workload.

I don't think this PR broke the native SSL support, because the single `netty-tcnative-classes` jar without corresponding native artifacts does not work.

**Pull request closed without merging**